src/nunit.analyzers/ValuesUsage/ValuesUsageAnalyzer.cs

```diff
@@ -0,0 +1,91 @@
+                                                                                                                title: ValuesUsageAnalyzerConstants.ParameterTypeMismatchTitle,
+                                                                                                                messageFormat: ValuesUsageAnalyzerConstants.ParameterTypeMismatchMessage,
+                                                                                                                category: Categories.Structure,
+                                                                                                                defaultSeverity: DiagnosticSeverity.Error,
+                                                                                                                description: ValuesUsageAnalyzerConstants.ParameterTypeMismatchDescription);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(parameterTypeMismatch);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterCompilationStartAction(AnalyzeCompilationStart);
+        }
+
+        private static void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var valuesType = context.Compilation.GetTypeByMetadataName(NUnitFrameworkConstants.FullNameOfTypeValuesAttribute);
+            if (valuesType is null)
+            {
+                return;
+            }
+
+            context.RegisterSymbolAction(symbolContext => AnalyzeParameter(symbolContext, valuesType), SymbolKind.Parameter);
+        }
+
+        private static void AnalyzeParameter(SymbolAnalysisContext symbolContext, INamedTypeSymbol valuesType)
+        {
+            var parameterSymbol = (IParameterSymbol)symbolContext.Symbol;
+            var attributes = parameterSymbol.GetAttributes();
+            if (attributes.Length == 0)
+            {
+                return;
+            }
+
+            foreach (var attribute in attributes.Where(x => x.ApplicationSyntaxReference is not null
+                                                       && SymbolEqualityComparer.Default.Equals(x.AttributeClass, valuesType)))
+            {
+                symbolContext.CancellationToken.ThrowIfCancellationRequested();
+                for (var index = 0; index < attribute.ConstructorArguments.Length; index++)
+                {
+                    var constructorArgument = attribute.ConstructorArguments[index];
+
+                    // If the compiler detects an illegal constant, we shouldn't check it.
+                    // Unfortunately the constant 'null' is marked as Error with a null type.
```

Comment:
@manfred-brands This seems incorrect - when I try using `null`, I see it as Kind == Primitive and Type as object (or Kind == Array and Type == object?[], etc.).  I copied this from the relevant code for TestCaseUsageAnalyzer.  I can remove this comment and the check `constructorArgument.Type is not null` if you want?

Reply:
@andrewimcclement I indeed don't get it in your analyzer.
It was added to `TestCaseUsageAnalyzer` for #510 and there I still get it for the tests with `[TestCase(null)]`
The test cases that still triggers this is: `AnalyzeWhenArgumentPassesNullToValueType`

![image](https://github.com/nunit/nunit.analyzers/assets/8655218/f7f90ea3-cb39-415a-850b-344b82a50fff)

The same test case in the `ValuesUsageAnalyzer` shows:
![image](https://github.com/nunit/nunit.analyzers/assets/8655218/f2c11d3f-c237-42b7-ab03-a2a0b8fd7b6c)

Investigating further, the root of the cause is the `AdjustArguments` call done in `TestCaseUsageAnalyzer`. 
This tries to reflect the case were the `TestCaseAttribute` constructor converts a `null` argument into `new object?[] { null }`.
Note that this is also the case for the `ValuesAttribute`.

I have converted the local  `AdjustArguments` into an extension method and call it from your analyzer.
I also added test cases to cover this scenario, although I don't see why anyone would. But it works in nunit.

Whilst I was there, I added support for nullable reference types with appropriate unit tests.